docs/src/main/asciidoc/security-keycloak-admin-client.adoc

```diff
@@ -85,16 +87,61 @@ public class RolesResource {
 }
 ----
 
-Now, you can produce the `Keycloak` bean yourself, or you can instruct Quarkus to produce `Keycloak` bean.
+<1> Create  a default Keycloak Admin Client which can perform Keycloak `master` realm administration tasks as an `admin-cli` client, such as adding new realms, clients and users.
+
+The only configuration which is required to create this Keycloak Admin Client is a Keycloak server URL, for example:
+
+[source,properties]
+----
+quarkus.keycloak.admin-client.server-url=http://localhost:8081
+----
+
+Injecting Keycloak Admin Client instead of creating it directly in the application code is a simpler and more flexible option but you can create the same admin client manually if necessary:
+
+[source,java]
+----
+package org.acme.keycloak.admin.client;
+
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.representations.idm.RoleRepresentation;
+
+import javax.annotations.PostConstruct;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.List;
+
+@Path("/api/admin")
+public class RolesResource {
+
+        Keycloak keycloak;
+
+        @PostConstruct
+        public void initKeycloak() {
+            keycloak = KeycloakBuilder.builder()
+                .serverUrl("http://localhost:8081")
+                .realm("master")
+                .clientId("admin-cli")
+                .username("admin")
+                .password("admin")
+                .build();
+        }
+
+        @GET
+        @Path("/roles")
+        public List<RoleRepresentation> getRoles() {
+            return keycloak.realm("quarkus").roles().list();
+        }
+
+}
+----
 
-NOTE: If you produce the bean yourself, please remember that `org.keycloak.admin.client.Keycloak` implements `AutoCloseable` and it's important to avoid resource exhaustion and related exceptions. If you configure your beans as explained below, Quarkus will do it for you.
+You can configure Keycloak admin client to administer other realms and clients. It can use either a `password` or `client_credentials` grant to acquire an access token to call Admin REST API which requires authorization.
 
-The Keycloak admin client needs an access token as its Admin REST API requires authorization.
 If you exchange user's credentials for the access token, here is an example configuration for the `password` grant type:
 
 [source,properties]
 ----
-quarkus.keycloak.admin-client=true
```

Comment:
If you wanna get rid of this flag, please adjust https://github.com/quarkusio/quarkus/blob/main/extensions/keycloak-admin-client-common/deployment/src/main/java/io/quarkus/keycloak/admin/client/common/KeycloakAdminClientInjectionEnabled.java#L11

Before you do so, please consider what's desired behavior - you mentioned in the docs only required property is `server-url`, but that's runtime property, so how are we going to determine whether user want us to provide the bean or not? Please adjust the logic if you drop the flag, thank you. I'm looking forward to see how you do it as the flag was best I got :-).

Reply:
@michalvavrik Sure, yeah, the extension features are enabled by default I believe so let me also change it to `true`

Reply:
Good. I didn't do it as making it true makes the change breaking (if the feature is enabled, it will require `server-url` during the app start), but it's pretty easy for users to disable it.

Reply:
@michalvavrik Oh I see, well, in that case I believe it has to be an optional string (server url), I'll update it

Reply:
Server URL is required by the validation `io.quarkus.keycloak.admin.client.common.KeycloakAdminClientConfigUtil#validate` but optional already, so when you drop the requirement there, everything will be fine till user tries to inject the client. Trade off is that user will only find out the URL is missing when the first call happens, but in this case it makes the feature user friendly. Thank you :-).

Reply:
@michalvavrik I'm thinking that if the server-url is not set then we  can doc that the injected `Keycloak` will throw ISA - I guess it should be reasonable as the only way the users will learn they can inject `Keycloak` is to read the guide/javaDocs, so they will be informed... Let me try and I'll update you once it is done

Reply:
@michalvavrik I've updated the code to return a null `Keycloak` if the `server-url` is not configured and doc-ed in docs/javadocs that in this case a `KeycloakBuilder` has to be used.
I've also updated the enabled property to be configured with the  `.enabled` - all other extensions have `.enabled`, without it it is also a bit tricky to have it configured all in YAML.

I'm not 100% happy with the idea of NPE if `server-url` is not configured - but we warn the users in the docs and show how to use an alternative `KeycloakBuilder` approach and having a feature enabled by default is consistent with the way other extensions handle it.

Does it sound reasonable ? Please double check the code as well

Thanks

Reply:
I can see now having a dedicated -cdi extension would be technically cleaner which you had, as with the current approach we can't fail OOB since users can just use `KeycloakBuilder`,  but it does feel it would be a bit too much / too many extensions at the same time, so IMHO the compromise is reasonable

Reply:
@michalvavrik Thanks, yeah, in the majority of cases an explicit `enabled` is used AFAIK. Thanks for the review so far